variable/trigonometry.cpp

```diff
@@ -30,16 +28,13 @@ Variable sin(Variable &&var) {
 
 Variable &sin(const Variable &var, Variable &out) {
   core::expect::unit_any_of(var, {units::rad, units::deg});
```

Comment:
Maybe this is redundant now since `to_unit` will throw in this case?

Reply:
In principle yes. But it provides some extra information to the user other than 'cannot be converted to rad'.

Reply:
I was wondering if, if removed, we could generate these functions of we extend the templates to support providing "pre-processors" for the inputs. In most cases there would be no pre-processor, here it would be the `to_unit` call.

Reply:
I had a similar idea but restricted to unit conversions. Instead of using just `to_unit` as a preprocessor, we could also use a function that produces a better error message similar to what we have now.